lua/DCS-gRPC/methods/net.lua

```diff
@@ -52,3 +52,18 @@ GRPC.methods.forcePlayerSlot = function(params)
 
     return GRPC.success(nil)
 end
+
+GRPC.methods.kickPlayer = function(params)
+  if params.id == 1 then
+    return GRPC.errorInvalidArgument("Cannot kick the server user")
+  end
+
+  local player_id = net.get_player_info(params.id, "id")
+
+  if not player_id then
+    return GRPC.errorNotFound("Could not find player with the ID of " .. params.id)
+  end
+
+  net.kick(params.id, params.message)
+  return GRPC.success(nil)
```

Comment:
I cannot actually get this API to return an error response for reasons I am not sure about (I am sure I am missing something) however it _does_ appear to exit the method, it just doesn't return an error response.

This is "good enough" at the moment and I don't want to hold up a release for it.

Reply:
The reason is that the Rust code does not wait on the response from the Lua function run.

Here is an example for you to reference. I did spot a few other API's which does not consider the error response as well, so we will have to do a sweep for them at some time in the future.

https://github.com/DCS-gRPC/rust-server/blob/2db3b3bf4b1be0f7628a8171356198770d59dd89/src/rpc/atmosphere.rs#L8-L18

Reply:
The `?` at the end of line 12 in the snippet above should be all that is needed to handle the error. I cannot spot anything wrong with the code. I'll test it once I am back on my PC (later tomorrow probably) to look into it.

Reply:
In my code I use the notification method because there is no response and I forgot that doing so would also swallow the error message. So there isn't anything for you to do here I think. I just need to fix the code I am using

Reply:
Ah gotcha, I misunderstood @justin-lovell example - it was an example where the error is handled. We can also remove the `self.notification()` method and use `self.request()` everywhere. Might be a good idea for a consistent behavior?

Reply:
I agree @rkusa -- if we do decide to have a async processor message box style in the future, and then the outcome of the notification should be read. At this moment, that API style feels too much overhead when most of the API's should be immediate request/response for simplicity.